### PR TITLE
Add more exceptions to the API

### PR DESCRIPTION
**Pull request by rockstar**
We've always known that our error handling has needed work. The original API over-caught things, and the new API doesn't catch anything. The new api's inability to catch things results in some really random errors (like, when `wait=True`, not finding the `operation` key in the returned dict).

We will likely need some sort of assertion on _every_ request we make, that makes sure we have the proper return statuses. We'll also need to think about the extra information we include in these new exceptions (a core reason why I try and use built-ins when I can).

### CODE REVIEW

**Comment by jpic** [comment]
Nice rewrite of #98, is it possible to have the response object too ?

File: pylxd/container.py [hunk 1]
```diff
@@ -65,7 +66,7 @@ def create(cls, client, config, wait=False):
         response = client.api.containers.post(json=config)
 
         if response.status_code != 202:
-            raise RuntimeError('Error creating instance')
+            raise exceptions.CreateFailed()
```
Comment (jpic) [inline pylxd/container.py]: Perhaps it would be useful to have the reason too, it's in the response object.

File: pylxd/exceptions.py [hunk 2]
```diff
@@ -1,2 +1,10 @@
 class ClientConnectionFailed(Exception):
     """An exception raised when the Client connection fails."""
+
+
+class NotFound(Exception):
+    """Generic get failure exception."""
+
+
+class CreateFailed(Exception):
+    """Generic create failure exception."""
```
Comment (jpic) [inline pylxd/exceptions.py]: Oh, no parent module-level exception ? I thought it was a best practice and helped in exception handling ...

File: pylxd/container.py (hunk 1 above)
Comment (rockstar) [inline pylxd/container.py]: I don't really want to use the whole response, though I agree we need more information. The response json's message param would seem appropriate as a good start. I'll add it.

**Comment by zulcss** [comment]
lgtm

**Pull request merged**